Neos.Flow/Classes/Http/RequestHandler.php

```diff
@@ -90,15 +95,15 @@ public function getPriority()
      */
     public function handleRequest()
     {
-        // Create the request very early so the ResourceManagement has a chance to grab it:
-        $request = Request::createFromEnvironment();
-        $response = new Response();
-        $this->componentContext = new ComponentContext($request, $response);
-
         $this->boot();
         $this->resolveDependencies();
+
+        $request = Request::createFromEnvironment();
+        $response = $this->responseFactory->createResponse();
         $response = $this->addPoweredByHeader($response);
-        $this->componentContext->replaceHttpResponse($response);
+        $this->componentContext = new ComponentContext($request, $response);
+        $this->bootstrap->getObjectManager()->setInstance(ComponentContext::class, $this->componentContext);
```

Comment:
Sneaky attempt to make `ComponentContext` injectable.

Reply:
yeah, no, please. I think we need to think very careful how we want to provide this, but not that way.